controllers/dspipeline_controller.go

```diff
@@ -501,6 +507,46 @@ func (r *DSPAReconciler) PublishMetrics(dspa *dspav1alpha1.DataSciencePipelinesA
 	}
 }
 
+func (r *DSPAReconciler) GetComponents(ctx context.Context, dspa *dspav1alpha1.DataSciencePipelinesApplication) dspav1alpha1.ComponentStatus {
```

Comment:
The way this is all currently written, there's no way to distinguish between "we got an error because the service/route is not up yet" (which is perfectly fine!) vs "we got an error because something bad happened while trying to read information from the service/route" (which is bad and we want to log an error!).

I suggest to rework this a bit to be able to differentiate those. I noticed that database.go has an `isDatabaseAccessible` method. I'd suggest adding something like that.

So roughly:

```
if isEnvoyRouteIsAvailable() {    
    err := GetEnvoyRouteHostname ...
    if err != nil {
        log.Error(err, "couldn't get hostname from mlmd proxy route")   <-- this is bad, so log an error
    }
}
```

and that applies to the service, and for the apiserver ones too

Reply:
imho Service/Route is available instantly after creation, but I believe you are talking about API Server and MLMD Proxy availability to serve requests, right? This should already be covered by the conditions.

Reply:
nope, I was talking about their existence. 

> Service/Route is available instantly after creation
typically yes, but kubernetes is non-deterministic because of the way controllers work. Relying on them to exist immediately after the manifests are applied opens us up to race conditions.

In other words, I'm suggesting there's two things to check for:

1 DSPO has applied the manifest to create the mlmd proy / envoy route, but it hasn't been stood up yet by kubernetes. This is an ok thing and just means we have to wait a bit and retry (i.e. wait for the Route controller to eventually make the world the way it should be).

2 DSPO has applied the manifest to create the mlmd proy / envoy route, it has been successfully created  by kubernetes, but we hit some problem trying to read information from it. This would likely indicate some bug in our code and we should throw an error here.